helpers/excel.js

```diff
@@ -140,6 +145,31 @@ async function loadDocuments(bim360, hubId, projectId) {
     return [].concat.apply([], results);
 }
 
+async function loadReferencedDocuments(bim360, projectId, issues) {
+    // Extract target_urn from issues array then filter null & duplicate urns 
+    const issuesTragetUrns = issues.map(issue => issue.target_urn);
+    const uniqueIssuesTargetUrns = issuesTragetUrns.filter((value, index, self) => value && self.indexOf(value) === index);
+
+    let referencedDocuments = [];
+    for (const targetUrn of uniqueIssuesTargetUrns) {
+        try {
+            console.log("Retrieving BIM360 Document details", targetUrn);
+            const document = await bim360.getItemDetails(projectId, targetUrn);
+            referencedDocuments.push(document);
+            //console.log(document);
+        } catch (err) {
+            console.error(err);
+            if (err.isAxiosError && err.response.status == 429) {  // Rate limit: Quota limit exceeded.
+                const retryAfter = parseInt(err.response.headers["retry-after"]);
+                if (retryAfter)
+                    await new Promise(resolve => setTimeout(resolve, retryAfter * 1000));
```

Comment:
I'm sorry, I didn't realize that any new commits in the same branch will be part of the pull request automatically, this is intended to be a WIP for another functionality that corrects an error when running the export function on a production BIM 360 account where the URNs collected by the loadDocuments function returns URNs that are not matching with the target_urn in the issues data (possibly the loadDocuments function returns the tip version of the files in the BIM 360 account, while the issues are linked to older versions that are not return from the search endpoint).
And you are correct, I realized the "skip instead of retry on 429" error after I committed, I've corrected this in the code but I'm testing it.
I'm also thinking of using the "ListItems" command instead, as it can process up to 50 urns in each request, so we can reach up to 15000 urns per minute instead of 300 elements each minute (current Forge Rate limit for both endpoints are 300 per minute).
Kindly disregard this commit, after I finish the function I'll resubmit (and correct the pending issue in the original pull request 🙂).

Reply:
Alright :)